From 3d45339f90ef9090bb6b00f53d796270a5fda770 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 20 Feb 2024 17:13:01 +0100
Subject: [PATCH 0679/1695] plat-stm32mp1: conf: default enable HASH

Default enable HASH compilation.
Enable CFG_STM32_CRYPTO_DRIVERS if any crypto IP is compiled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5762068790bd5055f9cba50593331aee580cdf86 plat-stm32mp1: conf: default enable HASH]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ifd6dfbe268cb905304dacbd25bc61d1db97330a0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361538
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 8f70a2b49..5d0fa4fb9 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -276,6 +276,7 @@ CFG_STM32_CPU_OPP ?= y
 CFG_STM32_CRYP ?= y
 CFG_STM32_ETZPC ?= y
 CFG_STM32_GPIO ?= y
+CFG_STM32_HASH ?= y
 CFG_STM32_I2C ?= y
 CFG_STM32_IWDG ?= y
 CFG_STM32_LPTIMER ?= y
@@ -316,7 +317,7 @@ endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
 endif # CFG_STM32MP13
 
 # If any crypto driver is enabled, enable the crypto-framework layer
-ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES),y)
+ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES CFG_STM32_HASH),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-- 
2.47.2

